compiler/main/driver.cpp

```diff
@@ -1074,6 +1076,11 @@ bool useDefaultEnv(std::string key) {
     }
   }
 
+  // Always use default env for internal variables that could include spaces
+  if (key == "CHPL_THIRD_PARTY_LINK_ARGS") {
+    return true;
+  }
+
```

Comment:
I don't think we need this anymore.... it was a workaround for CHPL_MAKE_ALL_VARS but now we don't even generate that anymore in the generated Makefile.

Reply:
Just to be clear, are you referring to the if block or the entire `useDefaultEnv` function?

Reply:
The block. I suspect we don't need useDefaultEnv either but am not ready to go that far yet. You added it in PR #2929, maybe you can figure out if we need it anymore.

I suspect that the factor out of CHPL_RUNTIME_ARCH will solve the issue.

Reply:
I'm pretty sure it can be removed, but let's save it for follow-up